src/Orleans.Core/CodeGeneration/TypeUtils.cs

```diff
@@ -556,24 +556,45 @@ public static IEnumerable<TypeInfo> GetDefinedTypes(Assembly assembly, Logger lo
             }
             catch (Exception exception)
             {
-                if (logger != null && logger.IsWarning)
-                {
-                    var message =
-                        $"Exception loading types from assembly '{assembly.FullName}': {LogFormatter.PrintException(exception)}.";
-                    logger.Warn(ErrorCode.Loader_TypeLoadError_5, message, exception);
-                }
-                
                 var typeLoadException = exception as ReflectionTypeLoadException;
+
                 if (typeLoadException != null)
                 {
+                    if (typeLoadException.LoaderExceptions != null)
+                    {
+                        //
+                        // If we've only BadImageFormatExceptions in LoaderExceptions, then it's ok to not to log, otherwise log
+                        // as a warning.
+                        //
+
+                        if (logger != null && logger.IsWarning)
+                        {
+                            var badImageFormatExceptionCount = typeLoadException.LoaderExceptions.Count(ex => ex as BadImageFormatException != null);
+
+                            if (typeLoadException.LoaderExceptions.Count() != badImageFormatExceptionCount)
```

Comment:
nit: `if (typeLoadException.LoaderExceptions.Any(ex => !(ex is BadImageFormatException))`